pkg/slayers/path/scion/raw.go

```diff
@@ -397,54 +398,71 @@ func (s *Raw) GetCurrentInfoField( /*@ ghost ubuf []byte @*/ ) (res path.InfoFie
 // @ ensures  sl.AbsSlice_Bytes(ubuf, 0, len(ubuf))
 // @ ensures  r != nil ==> r.ErrorMem()
 // posts for IO:
-// @ ensures  r == nil && idx == int(old(s.GetCurrINF(ubuf))) ==>
+// @ ensures  r == nil ==>
 // @ 	validPktMetaHdr(ubuf) && s.EqAbsHeader(ubuf)
 // @ ensures  r == nil && idx == int(old(s.GetCurrINF(ubuf))) ==>
 // @ 	let oldPkt := old(s.absPkt(ubuf)) in
-// @ 	let newPkt := AbsSetInfoField(oldPkt, info.ToIntermediateAbsInfoField()) in
+// @ 	let newPkt := io.UpdateInfoField(oldPkt, info.ToAbsInfoField()) in
```

Comment:
Maybe I would make `io.UpdateInfoField` a method, such that this could be written as 
```suggestion
// @ 	let newPkt := oldPkt.UpdateInfoField(info.ToAbsInfoField()) in
```